pkg/yttlibrary/url.go

```diff
@@ -27,13 +27,23 @@ var (
 
 				"query_params_encode": starlark.NewBuiltin("url.query_params_encode", core.ErrWrapper(urlModule{}.QueryParamsEncode)),
 				"query_params_decode": starlark.NewBuiltin("url.query_params_decode", core.ErrWrapper(urlModule{}.QueryParamsDecode)),
+
+				"parse": starlark.NewBuiltin("url.parse", core.ErrWrapper(urlModule{}.ParseURL)),
 			},
 		},
 	}
 )
 
 type urlModule struct{}
 
+// urlValue stores a parsed URL
+type urlValue struct {
+	url *url.URL
+	*core.StarlarkStruct
+}
+
+type urlUser urlValue
```

Comment:
_(optional)_

is `urlUser` _actually_ a whole `urlValue`?
Would it make sense for it to hold a `*url.UserInfo`?

Reply:
Sure, fixed.